README.md
# Fitness models

[![npm version](https://badge.fury.io/js/fitness-models.svg)](https://badge.fury.io/js/fitness-models)
[![renovate-app](https://img.shields.io/badge/renovate-app-blue.svg)](https://renovateapp.com/) 
[![Known Vulnerabilities](https://snyk.io/test/github/fabulator/fitness-models/badge.svg)](https://snyk.io/test/github/fabulator/fitness-models)
[![codecov](https://codecov.io/gh/fabulator/fitness-models/branch/master/graph/badge.svg)](https://codecov.io/gh/fabulator/fitness-models) 
[![travis](https://travis-ci.org/fabulator/fitness-models.svg?branch=master)](https://travis-ci.org/fabulator/fitness-models)


CHANGELOG.md
# Change Log
All notable changes to this project will be documented in this file.

## v8.1.0 (2022-08-14)
### Changed
- [e9c1ac0](https://github.com/fabulator/fitness-models/commit/e9c1ac038e66e2694c1f76bd7c7a69bff2718760) Bump dependencies

## v8.0.0 (2022-08-14)
### Changed
- [ae4e4ce](https://github.com/fabulator/fitness-models/commit/ae4e4ce9bde12a5c704318c8d76991ba8859db44) Update dependency mathjs to v11 BREAKING (#207)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

## v7.0.0 (2022-08-13)
### Changed
- [7b2b9f0](https://github.com/fabulator/fitness-models/commit/7b2b9f06d136d2779dfcfb675c9e93803c7f1046) Update dependency luxon to v3 BREAKING (#206)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
- [76e601b](https://github.com/fabulator/fitness-models/commit/76e601bcb3a187c13a73b92c84dc48b9938d17c0) Update dev dependency husky to v8  (#205)

Co-authored-by: Renovate Bot <bot@renovateapp.com>
- [48b44c9](https://github.com/fabulator/fitness-models/commit/48b44c92a8c9c43c27ec4ab17e20435629a95dc9) Update dependency gpx-builder to v5 BREAKING (#203)

Co-authored-by: Renovate Bot <bot@renovateapp.com>
- [f7dbd9c](https://github.com/fabulator/fitness-models/commit/f7dbd9c35d414f85d84d2b02d04832f3dc2535fe) Bump dev dependencies 
- [0d59dd4](https://github.com/fabulator/fitness-models/commit/0d59dd4e1b3add369bcc45b58f7c054110e222e0) Update actions/checkout action to v3 (#197)  

Co-authored-by: Renovate Bot <bot@renovateapp.com>
- [1a98879](https://github.com/fabulator/fitness-models/commit/1a988790ac65e4213d1b81ea0f56c8d803dc3384) Update actions/setup-node action to v3 (#196)  

Co-authored-by: Renovate Bot <bot@renovateapp.com>
- [117a31f](https://github.com/fabulator/fitness-models/commit/117a31f195a17b6245ff0b292de85e732fe7c382) Update dev dependency commitlint-config-fabulator to v3  (#195)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

## v6.0.0 (2022-02-06)
### Changed
- [907e91a](https://github.com/fabulator/fitness-models/commit/907e91aa9c3d4c87cc337a7f3c3761e1e23e08e3) Update dependency gpx-builder to v4 BREAKING (#194)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

## v5.0.0 (2022-02-06)
### Changed
- [a3c4eec](https://github.com/fabulator/fitness-models/commit/a3c4eec182f5142ca159893e4595be810c2391e3) Update dependency mathjs to v10 BREAKING (#190)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

## v4.2.0 (2021-09-30)
### Changed
- [b4a3cee](https://github.com/fabulator/fitness-models/commit/b4a3cee97ddcda2b73bbd0fe9077e4bcd3faed60) Bump dependencies

## v4.1.9 (2021-08-23)
### Fixed
- [6c6ee22](https://github.com/fabulator/fitness-models/commit/6c6ee22b09c728450d8516a01c99749f656e99d3) Update lock for security updated

## v4.1.8 (2021-08-16)
### Fixed
- [ddd1567](https://github.com/fabulator/fitness-models/commit/ddd156755dd31f4133f102a9c6932ee3f8564a9c) Update lock for security updated

## v4.1.7 (2021-08-09)
### Fixed
- [25442f5](https://github.com/fabulator/fitness-models/commit/25442f5660ae6774f79181087df6eee5203fe222) Update lock for security updated

## v4.1.6 (2021-08-02)
### Fixed
- [6f7df86](https://github.com/fabulator/fitness-models/commit/6f7df861f4d5bcf928b053a9386446df95755200) Update lock for security updated

## v4.1.5 (2021-08-02)
### Fixed
- [fb7578c](https://github.com/fabulator/fitness-models/commit/fb7578c679b0bdb28270dccb4400e3065c9420a1) Update lock for security updated

## v4.1.4 (2021-07-26)
### Fixed
- [69c9a0b](https://github.com/fabulator/fitness-models/commit/69c9a0b41cd6d2651efffc29533c50086f5a60cb) Update lock for security updated

## v4.1.3 (2021-07-25)
### Fixed
- [a9069e3](https://github.com/fabulator/fitness-models/commit/a9069e377d3cc098c714cc82f3af74f91836a439) Update lock for security updated

## v4.1.2 (2021-07-19)
### Fixed
- [4a47675](https://github.com/fabulator/fitness-models/commit/4a4767572630f49bd5bb8340f174e4fa9b7717da) Update lock for security updated

## v4.1.1 (2021-07-18)
### Fixed
- [d3b3ce3](https://github.com/fabulator/fitness-models/commit/d3b3ce353808decae422c9622d2499e51cdf4132) Update lock for security updated

## v4.1.0 (2021-07-18)
### Changed
- [287876c](https://github.com/fabulator/fitness-models/commit/287876ccbd2e28459c883c3d737b413fc846743a) Update dependencies

## v4.0.3 (2021-06-14)
### Fixed
- [e48f68b](https://github.com/fabulator/fitness-models/commit/e48f68b53e52cf0fbb7f97b264dfc676e55dcc49) Update lock for security updated

## v4.0.2 (2021-06-13)
### Fixed
- [6d3fbae](https://github.com/fabulator/fitness-models/commit/6d3fbae9723671af69099440166944ea934bc895) Update lock for security updated

## v4.0.1 (2021-06-07)
### Fixed
- [bd65b19](https://github.com/fabulator/fitness-models/commit/bd65b19f75300c14072d985eed4ebadcca613970) Update lock for security updated

## v4.0.0 (2021-06-06)
### Changed
- [8d3702f](https://github.com/fabulator/fitness-models/commit/8d3702f36d27513013413bddf8c0ae68a0815094) Update dependency mathjs to v9 BREAKING (#139)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

## v3.2.63 (2021-05-31)
### Fixed
- [6e73934](https://github.com/fabulator/fitness-models/commit/6e73934f05f3554954d537fdeabe263293163541) Update lock for security updated

## v3.2.62 (2021-05-24)
### Fixed
- [486eed7](https://github.com/fabulator/fitness-models/commit/486eed7d6ea58d9c61ed2cba1d0cb6c80fe52537) Update lock for security updated

## v3.2.61 (2021-05-24)
### Fixed
- [d41e812](https://github.com/fabulator/fitness-models/commit/d41e8121f64daa0e750a9e8b7499f42c4ab28ff8) Update lock for security updated

## v3.2.60 (2021-05-17)
### Fixed
- [7278f77](https://github.com/fabulator/fitness-models/commit/7278f77e59aee1e329a529630235ee2aa6b9c56a) Update lock for security updated

## v3.2.59 (2021-05-10)
### Fixed
- [0c3a100](https://github.com/fabulator/fitness-models/commit/0c3a1002735583f130a64141d746d14d45ade988) Update lock for security updated

## v3.2.58 (2021-05-03)
### Fixed

src/workout-types.ts
export const RUNNING = 0;
export const CYCLING_TRANSPORT = 1;
export const CYCLING_SPORT = 2;
export const MOUNTAIN_BIKINGS = 3;
export const SKATING = 4;
export const ROLLER_SKIING = 5;
export const SKIING_CROSS_COUNTRY = 6;
export const SKIING_DOWNHILL = 7;
export const SNOWBOARDING = 8;
export const KAYAKING = 9;
export const KITE_SURFING = 10;
export const ROWING = 11;
export const SAILING = 12;
export const WINDSURFING = 13;
export const FINTESS_WALKING = 14;
export const GOLFING = 15;
export const HIKING = 16;
export const ORIENTEERING = 17;
export const WALKING = 18;
export const RIDING = 19;
export const SWIMMING = 20;
export const CYCLING_INDOOR = 21;
export const OTHER = 22;
export const AEROBICS = 23;
export const BADMINTON = 24;
export const BASEBALL = 25;
export const BASKETBALL = 26;
export const BOXING = 27;
export const CLIMBING_STAIRS = 28;
export const CRICKET = 29;
export const ELLIPTICAL_TRAINING = 30;
export const DANCING = 31;
export const FENCING = 32;
export const FOOTBALL_AMERICAN = 33;
export const FOOTBALL_RUGBY = 34;
export const FOOTBALL_SOCCER = 35;
export const HANDBALL = 36;
export const HOCKEY = 37;
export const PILATES = 38;
export const POLO = 39;
export const SCUBA_DIVING = 40;
export const SQUASH = 41;
export const TABLE_TENIS = 42;
export const TENNIS = 43;
export const VOLEYBALL_BEACH = 44;
export const VOLEYBALL_INDOOR = 45;
export const WEIGHT_TRAINING = 46;
export const YOGA = 47;
export const MARTINAL_ARTS = 48;
export const GYMNASTICS = 49;
export const STEP_COUNTER = 50;
export const CIRKUIT_TRAINING = 87;
export const RUNNING_TREADMILL = 88;
export const SKATEBOARDING = 89;
export const SURFING = 90;
export const SNOWSHOEING = 91;
export const WHEELCHAIR = 92;
export const CLIMBING = 93;
export const WALKING_TREADMILL = 94;
export const KICK_SCOOTER = 95;
export const STAND_UP_PADDLING = 96;
export const TRAIL_RUNNING = 97;
export const ROWING_INDOORS = 98;
export const FLOORBALL = 99;
export const ICE_SKATING = 100;
export const SKI_TOURING = 101;
export const ROPE_JUMPING = 102;
export const STRETCHING = 103;
export const CANICROSS = 104;
export const PADDLE_TENNIS = 105;
export const PARAGLIDING = 106;

export type WorkoutType = typeof RUNNING |
    typeof CYCLING_TRANSPORT |
    typeof CYCLING_SPORT |
    typeof MOUNTAIN_BIKINGS |
    typeof SKATING |
    typeof ROLLER_SKIING |
    typeof SKIING_CROSS_COUNTRY |
    typeof SKIING_DOWNHILL |
    typeof SNOWBOARDING |
    typeof KAYAKING |
    typeof KITE_SURFING |
    typeof ROWING |
    typeof SAILING |
    typeof WINDSURFING |
    typeof FINTESS_WALKING |
    typeof GOLFING |
    typeof HIKING |
    typeof ORIENTEERING |
    typeof WALKING |
    typeof RIDING |
    typeof SWIMMING |
    typeof CYCLING_INDOOR |
    typeof OTHER |
    typeof AEROBICS |
    typeof BADMINTON |
    typeof BASEBALL |
    typeof BASKETBALL |
    typeof BOXING |
    typeof CLIMBING_STAIRS |
    typeof CRICKET |
    typeof ELLIPTICAL_TRAINING |
    typeof DANCING |
    typeof FENCING |
    typeof FOOTBALL_AMERICAN |
    typeof FOOTBALL_RUGBY |
    typeof FOOTBALL_SOCCER |
    typeof HANDBALL |
    typeof HOCKEY |
    typeof PILATES |
    typeof POLO |
    typeof SCUBA_DIVING |
    typeof SQUASH |
    typeof TABLE_TENIS |
    typeof TENNIS |
    typeof VOLEYBALL_BEACH |
    typeof VOLEYBALL_INDOOR |
    typeof WEIGHT_TRAINING |
    typeof YOGA |
    typeof MARTINAL_ARTS |
    typeof GYMNASTICS |
    typeof STEP_COUNTER |
    typeof CIRKUIT_TRAINING |
    typeof SKATEBOARDING |
    typeof CLIMBING |
    typeof KICK_SCOOTER |
    typeof CANICROSS |
    typeof FLOORBALL |
    typeof ICE_SKATING |
    typeof RUNNING_TREADMILL |
    typeof SURFING |
    typeof SNOWSHOEING |
    typeof WHEELCHAIR |
    typeof WALKING_TREADMILL |
    typeof STAND_UP_PADDLING |
    typeof TRAIL_RUNNING |
    typeof ROWING_INDOORS |
    typeof SKI_TOURING |
    typeof ROPE_JUMPING |
    typeof STRETCHING |
    typeof PADDLE_TENNIS |
    typeof PARAGLIDING;
